zerver/views/streams.py

```diff
@@ -867,9 +867,20 @@ def delete_in_topic(
         ).values_list("message_id", flat=True)
         messages = messages.filter(id__in=deletable_message_ids)
 
-    messages = messages.select_for_update(of=("self",))
-
-    do_delete_messages(user_profile.realm, messages)
+    # Topics can be large enough that this request will inevitably time out.
+    # In such a case, it's good for some progress to be accomplished, so that
+    # full deletion can be achieved by repeating the request. For that purpose,
+    # we delete messages in atomic batches, committing after each batch.
+    # TODO: Ideally this should be moved to the deferred_work queue.
+    batch_size = RETENTION_STREAM_MESSAGE_BATCH_SIZE
```

Comment:
Batch size of 1000 was proposed in https://github.com/zulip/zulip/issues/22821#issuecomment-1242385989 but this value here is `100` (890cafac11235b1f75d440db8106c85fae940faa). I can change this to be specifically 1000 if that makes more sense, though it felt more natural to use an existing constant

Reply:
Yeah, this is probably fine.